spec/controllers/v2/value_definitions_controller_spec.rb

```diff
@@ -32,108 +33,33 @@
-        let(:empty_security_guide) { FactoryBot.create(:v2_security_guide) }
-        let(:item) { FactoryBot.create(:v2_value_definition) }
-
-        it 'returns no data from empty security guide' do
-          get :index, params: { security_guide_id: empty_security_guide.id, parents: %i[security_guide] }
+    it_behaves_like 'collection', :security_guide do
+      let(:extra_params) { { security_guide_id: security_guide.id } }
+    end
 
-          expect(response_body_data).to be_empty
-        end
-      end
+    include_examples 'with metadata', :security_guide do
+      let(:extra_params) { { security_guide_id: security_guide.id } }
     end
 
-    context 'Unathorized' do
-      let(:rbac_allowed?) { false }
+    it_behaves_like 'paginable', :security_guide do
+      let(:extra_params) { { security_guide_id: security_guide.id } }
+    end
 
-      it 'responds with unauthorized status' do
-        get :index, params: { security_guide_id: security_guide.id, parents: %i[security_guide] }
+    it_behaves_like 'sortable', :security_guide do
+      let(:extra_params) { { security_guide_id: security_guide.id } }
+    end
 
-        expect(response).to have_http_status :forbidden
-      end
+    it_behaves_like 'searchable', :security_guide do
+      let(:extra_params) { { security_guide_id: security_guide.id } }
     end
   end
 
   describe 'GET show' do
-    let(:value_definition) { FactoryBot.create(:v2_value_definition) }
-    let(:security_guide) { value_definition.security_guide }
-
-    context 'Authorized' do
-      let(:rbac_allowed?) { true }
-
-      it 'returns value_definition by id' do
-        item = hash_including('data' => {
-                                'id' => value_definition.id,
-                                'type' => 'value_definition',
-                                **attributes.each_with_object({}) do |(key, value), obj|
-                                  obj[key.to_s] = value_definition.send(value)
-                                end
-                              })
-
-        get :show, params: {
-          security_guide_id: security_guide.id,
-          id: value_definition.id,
-          parents: %i[security_guide]
-        }
-
-        expect(response.parsed_body).to match(item)
-      end
-
-      context 'under incorrect parent security guide' do
-        let(:item) { FactoryBot.create(:v2_value_definition) }
-        let(:security_guide) { FactoryBot.create(:v2_security_guide) }
-
-        it 'returns not_found' do
-          get :show, params: {
-            security_guide_id: security_guide.id,
-            id: item.id,
-            parents: %i[security_guide]
-          }
+    let(:item) { FactoryBot.create(:v2_value_definition) }
+    let(:security_guide) { item.security_guide }
 
-          expect(response).to have_http_status :not_found
-        end
-      end
+    it_behaves_like 'resource', :security_guide do
+      let(:extra_params) { { security_guide_id: security_guide.id, id: item.id } }
```

Comment:
What happens if you extract this one a level above?

Reply:
I get an error complaining that no route matches my request.

Reply:
When changing the definition order of `extra_params`, it is a viable solution. I've changed the comment in the shared example to reflect this.